src/MOCKernel.cpp

```diff
@@ -0,0 +1,165 @@
+#include "MOCKernel.h"
+#include "TrackGenerator.h"
+
+/**
+ * @brief Constructor for the MOCKernel assigns default values
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+MOCKernel::MOCKernel(TrackGenerator* track_generator, int row_num) {
+  _count = 0;
+  _max_tau = track_generator->retrieveMaxOpticalLength();
+}
+
+
+/**
+ * @brief Constructor for the VolumeKernel assigns default values, calls
+ *        the MOCKernel constructor, and pulls refernces to FSR locks and FSR
+ *        volumes from the provided TrackGenerator.
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+VolumeKernel::VolumeKernel(TrackGenerator* track_generator, int row_num) :
+                           MOCKernel(track_generator, row_num) {
+  _FSR_locks = track_generator->getFSRLocks();
+
+  if (_FSR_locks == NULL)
+    log_printf(ERROR, "Unable to create a VolumeKernel without first creating "
+               "FSR locks");
+
+  _FSR_volumes = track_generator->getFSRVolumes();
+  _quadrature = track_generator->getQuadrature();
+  _weight = 0;
+}
+
+
+/**
+ * @brief Constructor for the CounterKernel assigns default values and calls
+ *        the MOCKernel constructor
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+CounterKernel::CounterKernel(TrackGenerator* track_generator, int row_num) :
+                           MOCKernel(track_generator, row_num) {}
```

Comment:
Add two spaces to make this continuation line match with the args in the line above.